components/kubearchive/development/kustomization.yaml

```diff
@@ -14,10 +15,168 @@ secretGenerator:
+      apiVersion: admissionregistration.k8s.io/v1
+      kind: MutatingWebhookConfiguration
+      metadata:
+        name: kubearchive-mutating-webhook-configuration
+        annotations:
+          service.beta.openshift.io/inject-cabundle: "true"
+  - patch: |-
+      apiVersion: admissionregistration.k8s.io/v1
+      kind: ValidatingWebhookConfiguration
+      metadata:
+        name: kubearchive-validating-webhook-configuration
+        annotations:
+          service.beta.openshift.io/inject-cabundle: "true"
+  # These patches solve Kube Linter problems
+  - patch: |-
+      apiVersion: apps/v1
+      kind: Deployment
+      metadata:
+        name: kubearchive-api-server
+        namespace: kubearchive
+      spec:
+        template:
+          spec:
+            containers:
+              - name: kubearchive-api-server
+                env:
+                - name: KUBEARCHIVE_OTEL_MODE
+                  value: enabled
+                - name: OTEL_EXPORTER_OTLP_ENDPOINT
+                  value: http://otel-collector:4318
+                securityContext:
+                  readOnlyRootFilesystem: true
+                  runAsNonRoot: true
+  - patch: |-
+      apiVersion: apps/v1
+      kind: Deployment
+      metadata:
+        name: kubearchive-operator
+        namespace: kubearchive
+        annotations:
+          ignore-check.kube-linter.io/unset-memory-requirements: >
+            "This is temporary to troubleshoot an issue"
+
+      spec:
+        template:
+          spec:
+            containers:
+              - name: manager
+                env:
+                - name: KUBEARCHIVE_OTEL_MODE
+                  value: enabled
+                - name: OTEL_EXPORTER_OTLP_ENDPOINT
+                  value: http://otel-collector:4318
+                securityContext:
+                  readOnlyRootFilesystem: true
+                  runAsNonRoot: true
+                ports:
+                  - containerPort: 8081
+                resources:
+                  $patch: replace
+                  limits:
+                    cpu: 500m
+                  requests:
+                    cpu: 10m
+                    memory: 256Mi
+
+  - patch: |-
+      apiVersion: apps/v1
+      kind: Deployment
+      metadata:
+        name: kubearchive-sink
+        namespace: kubearchive
+      spec:
+        template:
+          spec:
+            containers:
+              - name: kubearchive-sink
+                env:
+                - name: KUBEARCHIVE_OTEL_MODE
+                  value: enabled
+                - name: OTEL_EXPORTER_OTLP_ENDPOINT
+                  value: http://otel-collector:4318
+                securityContext:
+                  readOnlyRootFilesystem: true
+                  runAsNonRoot: true
+                resources:
+                  limits:
+                    cpu: 200m
+                    memory: 1Gi
+                  requests:
+                    cpu: 200m
+                    memory: 900Mi
```

Comment:
the development overlay is used in e2e tests. Do you think we can reduce deployments' `resources` in this overlay so we'll require smaller clusters during e2e tests?

Reply:
I reduced the memory for all deployments, currently we are sitting at ~30Mi on staging because we introduced a fix that solved the high memory usage.